addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -1559,7 +1559,10 @@ var SnippetEditor = Widget.extend({
             var hasOption = false;
             for (var key in self.styles) {
                 if (key === name || regex.test(key)) {
-                    self.styles[key].notify(ev.data.name, ev.data.data);
+                    const style = self.styles[key];
+                    self.trigger_up('snippet_edition_request', {exec: async () => {
+                        await style.notify(ev.data.name, ev.data.data);
+                    }});
```

Comment:
@qsm-odoo In master, can we be comfortable to actually `await` `notify`'s returned `Promise` - and also make sure its operations are run within the mutex ?

Reply:
No, I am not sure this is a good idea indeed. I remember trying to do that at some point and gave up (but can't remember why). Maybe now the code is more robust and would welcome it but I doubt it. To be honest, I think the "notify" system should be removed entirely and replaced by something better (not sure what), maybe that was the reason why I gave up. Maybe after OWL conversion?

What was merged in 14.0 (cc @detrouxdev) is a bit scary here, making such a high level function be able to return a promise is too much of a change to me :grimacing: Especially if it is not awaited anyway over there :shrug: Was there no other way? I guess we could still revert it while it is recent if there is another way or workaround :grimacing: Or at least not advertise the function as returning a promise and adding a big comment why this particular notify is an exception that should not be followed or something like that :wink:

Reply:
@qsm-odoo In stable we made sure to not change the signature of `notify` and not return a `Promise`. The change in the base `notify` is the removal of an incorrect `@returns {Promise}` to reduce risks of having code relying on returning a `Promise` from `notify`.

Reply:
Oh yes, my bad. Somehow I read that diff like you added the return of a Promise :shrug: Seems nice then :+1: